plugins/flytekit-greatexpectations/flytekitplugins/great_expectations/schema.py

```diff
@@ -0,0 +1,291 @@
+        lv: Literal,
+        expected_python_type: Type[GreatExpectationsType],
+    ) -> GreatExpectationsType:
+        if not (lv and lv.scalar and (lv.scalar.primitive or lv.scalar.schema or lv.scalar.blob)):
+            raise AssertionError("Can only validate a literal value")
+
+        # fetch the configuration
+        conf_dict = GreatExpectationsTypeTransformer.get_config(expected_python_type)[1].to_dict()
+
+        ge_conf = GreatExpectationsFlyteConfig(**conf_dict)
+
+        # file path for FlyteSchema and FlyteFile
+        temp_dataset = ""
+
+        # return value
+        return_dataset = ""
+
+        # FlyteSchema
+        if lv.scalar.schema:
+
+            # if data batch is to be genearated, skip copying the parquet file
+            if not ge_conf.is_runtime:
+                if not ge_conf.local_file_path:
+                    raise ValueError("local_file_path is missing!")
+
+                # copy parquet file to user-given directory
+                ctx.file_access.get_data(lv.scalar.schema.uri, ge_conf.local_file_path, is_multipart=True)
+
+                temp_dataset = os.path.basename(ge_conf.local_file_path)
+
+            def downloader(x, y):
+                ctx.file_access.get_data(x, y, is_multipart=True)
+
+            return_dataset = (
+                FlyteSchema(
+                    local_path=ctx.file_access.get_random_local_directory(),
+                    remote_path=lv.scalar.schema.uri,
+                    downloader=downloader,
+                    supported_mode=SchemaOpenMode.READ,
+                )
+                .open()
+                .all()
+            )
+
+        # FlyteFile
+        if lv.scalar.blob:
+            uri = lv.scalar.blob.uri
+
+            # check if the file is remote
+            if ctx.file_access.is_remote(uri):
+                if not ge_conf.local_file_path:
+                    raise ValueError("local_file_path is missing!")
+
+                if os.path.isdir(ge_conf.local_file_path):
+                    local_path = os.path.join(ge_conf.local_file_path, os.path.basename(uri))
+                else:
+                    local_path = ge_conf.local_file_path
+
+                # download the file into local_file_path
+                ctx.file_access.get_data(
+                    remote_path=uri,
+                    local_path=local_path,
+                )
+
+            temp_dataset = os.path.basename(uri)
+
+            return_dataset = FlyteFile(uri)
+
+        if lv.scalar.primitive:
+            dataset = return_dataset = lv.scalar.primitive.string_value
+        else:
+            dataset = temp_dataset
+
+        batch_request_conf = ge_conf.batch_request_config
+
+        # fetch the data context
+        context = ge.data_context.DataContext(ge_conf.context_root_dir)
+
+        # minimalistic batch request
+        final_batch_request = {
+            "data_asset_name": "random_string" if ge_conf.is_runtime else dataset,
```

Comment:
Could we enable the user to pass this in to maintain the utility of `"data_asset_name"` instead of hard-coding it is "random_string"?

Reply:
`data_asset_name` is not shown on the user-facing side. It's palpable only within the plugin code. When the user sends a dataset that needs to be validated, the plugin populates either `data_asset_name` or `batch_data`/`query`. I've used "random_string" because `data_asset_name` doesn't have any real meaning when using a `RuntimeBatchRequest`. Giving a separate `data_asset_name` can be confusing; what do you think?

Reply:
Hi @samhita-alla - `data_asset_name` is shown when generating a data docs site. This helps users to track what data_asset was used for validation. You can see an example here:
<img width="1565" alt="Screen Shot 2021-08-09 at 11 10 05 AM" src="https://user-images.githubusercontent.com/15267635/128729498-914c16c7-b1a5-4594-870b-ec80ea2b2f5d.png">

If this is hard-coded, then all results will look the same, which could be confusing for users. The idea is that when users validate the same data asset multiple times (like monthly sales data for instance), they would use the same `data_asset_name`, and when they validate new data (like monthly user data), they would use a new `data_asset_name`. This isn't something that we can infer, but it is also something that improves the tracking of validation results over time, which is why we require user input.

Reply:
@talagluck: I've added a `data_asset_name` parameter.